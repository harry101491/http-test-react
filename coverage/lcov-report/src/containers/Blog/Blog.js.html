<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/containers/Blog/Blog.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/containers/Blog</a> Blog.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.54% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/13</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">37.5% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/8</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">61.54% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/13</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
&nbsp;
import API from '../../api';
&nbsp;
import Post from '../../components/Post/Post';
import FullPost from '../../components/FullPost/FullPost';
import NewPost from '../../components/NewPost/NewPost';
import Header from '../../components/Header/Header';
&nbsp;
import './Blog.scss';
&nbsp;
class Blog extends Component {
&nbsp;
    state = {
        posts: [],
        selectedPostId: null,
        error: false,
    };
    
    componentDidMount() {
        API
          .get('posts/')
          .then(posts =&gt; {
            const newPosts = posts.data.slice(0, 4);
            const updatedPosts = newPosts.map(<span class="fstat-no" title="function not covered" >(p</span>ost) =&gt; {
<span class="cstat-no" title="statement not covered" >                return {</span>
                    ...post,
                    author: 'Harshit',
                };
            });
&nbsp;
            this.setState({
              ...this.state,
              posts: updatedPosts,
            });
&nbsp;
          })
          .catch(<span class="fstat-no" title="function not covered" >()</span> =&gt; {
<span class="cstat-no" title="statement not covered" >              this.setState({</span>
                  error: true
              });
          });
    }
&nbsp;
    postClickedHandler = <span class="fstat-no" title="function not covered" >(i</span>d) =&gt; {
<span class="cstat-no" title="statement not covered" >        this.setState({</span>
            ...this.state,
            selectedPostId: id,
        });
    }
    
    render () {
        let posts = &lt;p style={{ textAlign: 'center' }}&gt; Something went wrong! &lt;/p&gt;;
&nbsp;
        <span class="missing-if-branch" title="else path not taken" >E</span>if(!this.state.error) {
            posts = this.state.posts
                .map(<span class="fstat-no" title="function not covered" >po</span>st =&gt; 
<span class="cstat-no" title="statement not covered" >                    &lt;Post </span>
                        key={post.id} 
                        {...post} 
                        clicked={<span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" >this.postClickedHandler(post.id)}</span>
                    /&gt;
                )
        }
        return (
            &lt;div&gt;
                &lt;Header /&gt;
                &lt;section className="Posts"&gt;
                    {posts}
                &lt;/section&gt;
                &lt;section&gt;
                    &lt;FullPost 
                        id={this.state.selectedPostId}
                    /&gt;
                &lt;/section&gt;
                &lt;section&gt;
                    &lt;NewPost /&gt;
                &lt;/section&gt;
            &lt;/div&gt;
        );
    }
}
&nbsp;
export default Blog;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jul 27 2019 23:59:10 GMT-0700 (Pacific Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
