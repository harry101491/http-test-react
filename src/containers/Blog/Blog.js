import React, { Component } from 'react';

import axios from 'axios';

import Post from '../../components/Post/Post';
import FullPost from '../../components/FullPost/FullPost';
import NewPost from '../../components/NewPost/NewPost';
import Header from '../../components/Header/Header';

import './Blog.scss';

class Blog extends Component {

    state = {
        posts: [],
        selectedPostId: null,
        error: false,
    };
    
    componentDidMount() {
        axios
          .get('posts/')
          .then(posts => {
            const newPosts = posts.data.slice(0, 4);
            const updatedPosts = newPosts.map((post) => {
                return {
                    ...post,
                    author: 'Harshit',
                };
            });

            this.setState({
              ...this.state,
              posts: updatedPosts,
            });

          })
          .catch(() => {
              this.setState({
                  error: true
              });
          });
    }

    postClickedHandler = (id) => {
        this.setState({
            ...this.state,
            selectedPostId: id,
        });
    }
    
    render () {
        let posts = <p style={{ textAlign: 'center' }}> Something went wrong! </p>;

        if(!this.state.error) {
            posts = this.state.posts
                .map(post => 
                    <Post 
                        key={post.id} 
                        {...post} 
                        clicked={() => this.postClickedHandler(post.id)}
                    />
                )
        }
        return (
            <div>
                <Header />
                <section className="Posts">
                    {posts}
                </section>
                <section>
                    <FullPost 
                        id={this.state.selectedPostId}
                    />
                </section>
                <section>
                    <NewPost />
                </section>
            </div>
        );
    }
}

export default Blog;