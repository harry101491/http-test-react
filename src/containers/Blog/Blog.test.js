import React from 'react';
import { shallow } from 'enzyme';
// import mockAxios from 'axios';

import Blog from './Blog';



describe('<Blog />', () => {

    let wrappper;

    beforeEach(() => {
        wrappper = shallow(<Blog />);
    });

    it('should render as expected', () => {
        expect(wrappper).toMatchSnapshot();
    });

    // it('componentDidMount() should call backend API', () => {
    //     const instance = wrappper.instance();
        
    //     instance.componentDidMount();

    //     expect(mockAxios.get).toHaveBeenCalledTimes(1);
    // });
});
