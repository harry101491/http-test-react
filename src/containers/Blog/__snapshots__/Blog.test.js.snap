// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Blog /> should render as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Blog />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="Posts"
>
          
</section>,
        <section>
          <FullPost
                    id={null}
          />
</section>,
        <section>
          <NewPost />
</section>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "Posts",
        },
        "ref": null,
        "rendered": Array [],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <FullPost
            id={null}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "id": null,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NewPost />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "section",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="Posts"
>
            
</section>,
          <section>
            <FullPost
                        id={null}
            />
</section>,
          <section>
            <NewPost />
</section>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "Posts",
          },
          "ref": null,
          "rendered": Array [],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <FullPost
              id={null}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NewPost />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "section",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
